project/frontend/src/components/PreferenceForm.js

```diff
@@ -56,6 +58,10 @@ class PreferenceForm extends React.Component {
     this.setState({ [attrName]: field });
   }
 
+  getLocationData(data) {
```

Comment:
does this return something? why is this a getter function? It seems like it's acting like a setter

Reply:
yep not really a clear name since its kind of doing both but its removed in the latest commit

Reply:
It's getting the data from the LocationFinder but also setting the state, if that makes sense... I agree it is more of a setter though